temperature/crud.py

```diff
@@ -0,0 +1,54 @@
+import datetime
+
+from sqlalchemy import select
+from sqlalchemy.orm import selectinload
+
+from city.crud import get_all_cities
+from temperature import models, schemas
+
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from temperature.utils import get_temperature
+
+
+async def update_temperatures(db: AsyncSession) -> dict:
+    cities = await get_all_cities(db=db)
+
+    temperatures = [
+        models.Temperature(
+            date_time=datetime.datetime.now(),
+            city_id=city.id,
+            temperature=await get_temperature(city.name)
+        )
+        for city in cities
+    ]
+    db.add_all(temperatures)
+
+    await db.commit()
+
+    return {"detail": "All temperatures are successfully updated"}
+
+
+async def get_all_temperatures(
+    db: AsyncSession,
+    city_id: int | None
```

Comment:
The type hint `int | None` is used here. Ensure that your Python version supports this syntax, or consider using `Optional[int]` from the `typing` module for compatibility with older versions.